### PR TITLE
Add additional settings to handle CORS

### PR DESCRIPTION
**Pull request by retroinspect**
- `CorsInterceptor` in frontend & `Access-Control-Allow-Credentials` in cloud orchestrator allows HTTP requests to carry cookies
- Explicitly added allowed methods 
- Handle preflight request as specified from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)

### CODE REVIEW

File: pkg/app/app.go [hunk 1]
```diff
@@ -122,6 +126,13 @@ func (c *App) Handler() http.Handler {
 	rootRouter.PathPrefix("/").Handler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		c.AddCorsHeaderIfNeeded(w, r)
 		router.ServeHTTP(w, r)
```
Comment (retroinspect) [inline pkg/app/app.go:128]: TODO: should be removed

Comment (retroinspect) [reply pkg/app/app.go]: Done

**Comment by retroinspect** [comment]
@k311093 PTAL

File: pkg/app/app.go [hunk 2]
```diff
@@ -121,7 +125,13 @@ func (c *App) Handler() http.Handler {
 	rootRouter := mux.NewRouter()
 	rootRouter.PathPrefix("/").Handler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		c.AddCorsHeaderIfNeeded(w, r)
-		router.ServeHTTP(w, r)
+		if r.Method == "OPTIONS" {
+			w.Header().Add("Allow", allowedMethods)
```
Comment (k311093) [inline pkg/app/app.go:129]: How about using r.Method instead of allowedMethods since it also allows the methods that is not requested.

**Review by k311093 — CHANGES_REQUESTED** [review]

File: pkg/app/app.go (hunk 2 above)
Comment (retroinspect) [reply pkg/app/app.go:129]: According to the [specification](https://httpwg.org/specs/rfc9110.html#field.allow), `Allow` header contains all request methods the server supports.
 While `r.Method` here always results in "OPTIONS", I think `allowedMethods` would be appropriate for the spec.

Comment (k311093) [reply pkg/app/app.go:129]: Acknowledged, thanks!

**Pull request merged**